### ISSUE TITLE
`control-has-associated-label` accepts whitespace as a label

### ISSUE THREAD

**Issue by wobo-nate** [jsx-eslint/eslint-plugin-jsx-a11y#918]
## The problem
The `control-has-associated-label` rule currently accepts whitespace as an accessible label. This issue is exaggerated by the use of `prettier` which may introduce line breaks and other whitespace automatically, defeating the rule.

### Examples
All of the below examples lack accessible labels.
```html
<button></button>                <!-- rule correctly flags invalid markup -->
<button> </button>               <!-- rule fails to flag invalid markup -->

<button aria-label=""></button>  <!-- rule correctly flags invalid markup -->
<button aria-label=" "></button> <!-- rule fails to flag invalid markup -->
```

## Suggested fix
Update `mayHaveAccessibleLabel` to `.trim()` the node's value before asserting truthiness.

**Comment by ljharb** [comment]
Seems a reasonable bugfix; anyone relying on it shouldn't be :-)
